File: frontend/src/services/customerService.js
import api from "./api";

export const getAllCustomers = () =>
    api.get("/admin/customers");

export const getCustomerById = (id) =>
    api.get(`/admin/customers/${id}`);

export const createCustomer = (customer) =>
    api.post("/admin/customers", customer);

export const updateCustomer = (id, customer) =>
    api.put(`/admin/customers/${id}`, customer);

export const deleteCustomer = (id) =>
    api.delete(`/admin/customers/${id}`);

File: backend/src/main/java/com/shiptrack/backend/service/CustomerService.java
package com.shiptrack.backend.service;

import com.shiptrack.backend.entity.Customer;

import java.util.List;

public interface CustomerService {

    List<Customer> getAllCustomers();

    Customer getCustomerById(Long id);

    Customer createCustomer(Customer customer);

    Customer updateCustomer(Long id, Customer customer);

    void deleteCustomer(Long id);

}